eng/cake/dotnet.cake

```diff
@@ -291,11 +296,11 @@ Task("dotnet-diff")
         else
         {
             // clean all working folders
-            var diffCacheDir = tempDirectory.Combine("diffCache");
+            var diffCacheDir = GetTempDirectory().Combine("diffCache");
             EnsureDirectoryExists(diffCacheDir);
             CleanDirectories(diffCacheDir.FullPath);
-            EnsureDirectoryExists(diffDirectory);
-            CleanDirectories(diffDirectory.FullPath);
+            EnsureDirectoryExists(GetDiffDirectory());
+            CleanDirectories(GetDiffDirectory().FullPath);
 
```

Comment:
I would maybe save the result of `GetDiffDirectory()` in a variable, so you don't have to call it twice.